tests/unittests/storage/test_storage.py

```diff
@@ -0,0 +1,363 @@
+
+            count = 0
+            for trial in cfg.trials:
+                if trial['status'] in {'new', 'suspended', 'interrupted'}:
+                    count += 1
+
+            assert len(trials) == count
+            for trial in trials:
+                assert trial.status in {'new', 'suspended', 'interrupted'}
+
+    def test_fetch_noncompleted_trials(self, storage):
+        """Test fetch non completed trials"""
+        with OrionState(
+                experiments=[base_experiment], trials=generate_trials(), database=storage) as cfg:
+            storage = cfg.storage()
+
+            experiment = cfg.get_experiment('default_name', 'default_user', version=None)
+            trials = storage.fetch_noncompleted_trials(experiment)
+
+            count = 0
+            for trial in cfg.trials:
+                if trial['status'] != 'completed':
+                    count += 1
+
+            assert len(trials) == count
+            for trial in trials:
+                assert trial.status != 'completed'
+
+    def test_fetch_completed_trials(self, storage):
+        """Test fetch completed trials"""
+        with OrionState(
+                experiments=[base_experiment], trials=generate_trials(), database=storage) as cfg:
+            count = 0
+            for trial in cfg.trials:
+                if trial['status'] == 'completed':
+                    count += 1
+
+            storage = cfg.storage()
+            experiment = cfg.get_experiment('default_name', 'default_user', version=None)
+            trials = storage.fetch_trial_by_status(experiment, 'completed')
+
+            assert len(trials) == count
+            for trial in trials:
+                assert trial.status == 'completed', trial
+
+    def test_count_completed_trials(self, storage):
+        """Test count completed trials"""
+        with OrionState(
+                experiments=[base_experiment], trials=generate_trials(), database=storage) as cfg:
+            count = 0
+            for trial in cfg.trials:
+                if trial['status'] == 'completed':
+                    count += 1
+
+            storage = cfg.storage()
+
+            experiment = cfg.get_experiment('default_name', 'default_user', version=None)
+            trials = storage.count_completed_trials(experiment)
+            assert trials == count
+
+    def test_count_broken_trials(self, storage):
+        """Test count broken trials"""
+        with OrionState(
+                experiments=[base_experiment], trials=generate_trials(), database=storage) as cfg:
+            count = 0
+            for trial in cfg.trials:
+                if trial['status'] == 'broken':
+                    count += 1
+
+            storage = cfg.storage()
+
+            experiment = cfg.get_experiment('default_name', 'default_user', version=None)
+            trials = storage.count_broken_trials(experiment)
+
+            assert trials == count
+
+    def test_update_heartbeat(self, storage):
```

Comment:
Should test with a trial that was set to something else than `reserved` meanwhile.

Reply:
the update of heartbeat does not depend on the status anymore

Reply:
It does seem to: https://github.com/Epistimio/orion/pull/250/files#diff-59f01e95e125eaa93bb5ab10115477f7R281

Reply:
that is after the update I made to make you happy